dissect/target/plugins/os/windows/thumbcache.py

```diff
@@ -0,0 +1,81 @@
+from typing import Iterator
+
+from dissect.thumbcache.thumbcache import Thumbcache
+
+from dissect.target.exceptions import UnsupportedPluginError
+from dissect.target.helpers.fsutil import TargetPath
+from dissect.target.helpers.record import TargetRecordDescriptor
+from dissect.target.plugin import Plugin, export
+
+GENERIC_THUMBCACHE_FIELDS = [
+    ("string", "identifier"),
+    ("string", "hash"),
+    ("string", "extension"),
+    ("varint", "data_size"),
+    ("bytes", "header_checksum"),
+    ("bytes", "data_checksum"),
+    ("uri", "path"),
+]
+IndexRecord = TargetRecordDescriptor(
+    "thumbcache/index",
+    [
+        ("string", "identifier"),
+        ("boolean", "in_use"),
+        ("varint", "flags"),
+        ("datetime", "last_modified"),
+        ("uri", "path"),
+    ],
+)
+ThumbcacheRecord = TargetRecordDescriptor("thumbcache/thumbcache", GENERIC_THUMBCACHE_FIELDS)
+IconcacheRecord = TargetRecordDescriptor("thumbcache/iconcache", GENERIC_THUMBCACHE_FIELDS)
+
+
+class ThumbcachePlugin(Plugin):
+    __namespace__ = "thumbcache"
+
+    def get_cache_paths(self) -> Iterator[TargetPath]:
+        for user_details in self.target.user_details.all_with_home():
+            cache_path = user_details.home_path / "appdata/local/microsoft/windows/explorer"
+            if cache_path.exists():
+                yield cache_path
+
+    def check_compatible(self) -> None:
+        for cache_path in self.get_cache_paths():
+            if len(list(cache_path.glob("*_idx.db"))) > 0:
+                return None
+        raise UnsupportedPluginError("There was no cache path for that plugin")
+
+    def plugin(self, record_type: TargetRecordDescriptor, prefix: str):
```

Comment:
`plugin` is maybe a bit too generic of a function name. Maybe `_parse` or something?